common/beerocks/hostapd/configuration.cpp

```diff
@@ -16,8 +19,122 @@ Configuration::Configuration(const std::string file_name) : m_configuration_file
 
 Configuration::operator bool() const { return m_ok; }
 
-void Configuration::load() { m_ok = true; }
+void Configuration::load() { 
+    // please take a look at the end of this file for
+    // the expected format of hostapd configuration file
+    // loading the file relays on the expected format
+    // otherwise the load fails
+
+    std::ifstream ifs(m_configuration_file);
+    std::string line;
+
+    bool parsing_vaps = false;
+    std::string cur_vap;
+
+    while (getline(ifs, line)) {
+
+        // skip empty lines
+        if (std::all_of(line.begin(),line.end(),isspace)){
+            continue;
+        }
+        // check if the string belongs to a vap config part and capture which one.
+        const std::string bss_eq("bss=");
+        if (line.compare(0, bss_eq.length(), bss_eq) == 0) {
+
+            // from now on we are in the vaps area, all 
+            // key/value pairs belongs to vaps
+            parsing_vaps = true;
+
+            // copy the bss (vap) value
+            cur_vap.assign(line, bss_eq.length(), std::string::npos);
+        }
+
+        // if not a vap line store it in the header part of the config,
+        // otherwise add to the currently being parsed vap storage.
+        if (!parsing_vaps) {
+            m_hostapd_config_head.push_back(line);
+        } else {
+            m_hostapd_config_vaps[cur_vap].push_back(line);
+        }
+    }
+
+    // log something if there was an error during the read
+    if (ifs.bad()) {
+        LOG(ERROR) << "Unable to read " << m_configuration_file << ": " << strerror(errno);
+    }
+
+    // if we've got to parsing vaps and no read errors, assume all is good
+    m_ok = parsing_vaps && !ifs.bad();
+}
+
+std::ostream& operator<<(std::ostream& o, const Configuration& conf) {
```

Comment:
If you make this
```suggestion
template<class T> T& operator<<(T& o, const Configuration& conf) {
```
it will also work as `LOG(DEBUG) << conf;`

Reply:
[mimic :scream:](https://github.com/prplfoundation/prplMesh/blob/master/framework/tlvf/src/include/tlvf/tlvftypes.h#L71)

Reply:
Hm, I wrote that code, so I guess it didn't work with the template since I'm sure I considered it at the time...